File: backend/src/main/java/org/sport/backend/entity/BankAccount.java
package org.sport.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;
import org.sport.backend.dto.base.BaseEntity;

import java.util.UUID;

@Entity
@Table(name = "bank_accounts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class BankAccount extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "bank_account_id")
    private UUID bankAccountId;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @Column(name = "bank_name", nullable = false, length = 100)
    private String bankName;

    @Column(name = "account_number", nullable = false, length = 50)
    private String accountNumber;

    @Column(name = "account_holder_name", nullable = false, length = 200)
    private String accountHolderName;

    @Column(name = "branch_name", length = 200)
    private String branchName;

    @Column(name = "is_verified")
    private Boolean isVerified = false;

    @Column(name = "verification_code", length = 50)
    private String verificationCode;

    private String qrCode;
}

File: mobile/lib/models/bank_account.dart
class BankAccountResponse {
  final String bankAccountId;
  final String bankName;
  final String accountNumber;
  final String accountHolderName;
  final String branchName;
  final String? qrCode;
  final bool? isVerified;

  BankAccountResponse({
    required this.bankAccountId,
    required this.bankName,
    required this.accountNumber,
    required this.accountHolderName,
    required this.branchName,
    this.qrCode,
    this.isVerified,
  });

  factory BankAccountResponse.fromJson(Map<String, dynamic> json) {
    return BankAccountResponse(
      bankAccountId: json['bankAccountId']?.toString() ?? '',
      bankName: json['bankName']?.toString() ?? '',
      accountNumber: json['accountNumber']?.toString() ?? '',
      accountHolderName: json['accountHolderName']?.toString() ?? '',
      branchName: json['branchName']?.toString() ?? '',
      qrCode: json['qrCode']?.toString(),
      isVerified: json['isVerified'] as bool?,
    );
  }
}